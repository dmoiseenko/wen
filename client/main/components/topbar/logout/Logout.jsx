import React from 'react';
import PropTypes from 'prop-types';


export default function Logout({ logout }) {
  function handleClick() {
    logout();
  }

  return (
    <a
      className="button is-primary"
      onClick={handleClick}
    >
      Log out
    </a>
  );
}

Logout.propTypes = {
  logout: PropTypes.func.isRequired
};
