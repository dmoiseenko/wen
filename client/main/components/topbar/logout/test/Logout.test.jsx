import React from 'react';
import { shallow } from 'enzyme';

import Logout from '../Logout';


it('should render', () => {
  const props = {
    logout: jest.fn()
  };

  const wrapper = shallow(
    <Logout {...props} />
  );

  expect(wrapper.node).toMatchSnapshot();
});

it('should call logout when click fires', () => {
  const logout = jest.fn();

  const wrapper = shallow(
    <Logout
      logout={logout}
    />
  );

  wrapper.find('.button').simulate('click');

  expect(logout).toHaveBeenCalled();
});
