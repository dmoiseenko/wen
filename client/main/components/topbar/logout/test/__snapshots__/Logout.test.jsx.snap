// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<a
  className="button is-primary"
  onClick={[Function]}
>
  Log out
</a>
`;
